src/XrdOssStats/XrdOssStatsFileSystem.cc

```diff
@@ -0,0 +1,271 @@
+
+#include "XrdOuc/XrdOucGatherConf.hh"
+#include "XrdOssStatsConfig.hh"
+#include "XrdOssStatsDirectory.hh"
+#include "XrdOssStatsFile.hh"
+#include "XrdOssStatsFileSystem.hh"
+#include "XrdSys/XrdSysPthread.hh"
+#include "XrdXrootd/XrdXrootdGStream.hh"
+
+#include <inttypes.h>
+#include <stdexcept>
+#include <thread>
+
+StatsFileSystem::StatsFileSystem(XrdOss *oss, XrdSysLogger *lp, const char *configfn, XrdOucEnv *envP) :
+    XrdOssWrapper(*oss),
+    m_oss(oss),
+    m_env(envP),
+    m_log(lp, "fsstat_"),
+    m_slow_duration(std::chrono::seconds(1))
+{
+    m_log.Say("------ Initializing the storage statistics plugin.");
+    if (!Config(configfn)) {
+        throw std::runtime_error("Failed to configure the storage statistics plugin.");
+    }
+    pthread_t tid;
+    int rc;
+    if ((rc = XrdSysThread::Run(&tid, StatsFileSystem::AggregateBootstrap, static_cast<void *>(this), 0, "FS Stats Compute Thread"))) {
+      m_log.Emsg("StatsFileSystem", rc, "create stats compute thread");
+      throw std::runtime_error("Failed to create the statistics computing thread.");
+    }
+
+    // While the plugin _does_ print its activity to the debugging facility (if enabled), its relatively useless
+    // unless the g-stream is available.  Hence, if it's _not_ available, we should fail to startup.
+    if (envP) {
+       m_gstream = reinterpret_cast<XrdXrootdGStream*>(envP->GetPtr("oss.gStream*"));
+       if (m_gstream) {
+         m_log.Say("Config", "Stats monitoring has been configured via xrootd.mongstream directive");
+       } else {
+         throw std::runtime_error("XrdOssStats plugin is loaded but it requires the oss monitoring g-stream to also be enabled to be set; try adding `xrootd.mongstream oss ...` to your configuration");
```

Comment:
Hmmm, this prevents me from having a base configuration and able to turn this monitoring on/off via mongstream directive. Instead I have to comment out the plugin stacking to turn stuff off which isn't particularly intuitive. Why not issue a message and return the underneath oss pointer which effectively doesn't load the plugin (well use it anyway). Plus, we do not use throw in the server unless there is no alternative. In your case, if you really want to kill server initialization return a nil pointer. Anyway, this test should be moved to XrdOssAddStorageSystem2() where you can properly handle this case, as described above, without using a throw.